x/distribution/types/expected_keepers.go

```diff
@@ -46,8 +46,9 @@ type StakingKeeper interface {
 	Validator(sdk.Context, sdk.ValAddress) stakingtypes.ValidatorI            // get a particular validator by operator address
 	ValidatorByConsAddr(sdk.Context, sdk.ConsAddress) stakingtypes.ValidatorI // get a particular validator by consensus address
 
-	// slash the validator and delegators of the validator, specifying offence height, offence power, and slash fraction
-	Slash(sdk.Context, sdk.ConsAddress, int64, int64, sdk.Dec)
+	// slash the validator and delegators of the validator, specifying offence height, offence power, slash fraction and
```

Comment:
A boolean doesn't really indicate offense type. Can we use an enum instead?

Reply:
You can follow `BOND_STATUS` for how to do this in proto and stuff in case you're curious

Reply:
Great idea @alexanderbez !

A proto enum would make things more clear and reusable here.

Reply:
@alexanderbez Since the slashing module has a staking module dependency,
the `InfractionType` enum should be defined in the `staking.proto` to avoid import cycle.

What's your take on this  ?